superset/viz.py

```diff
@@ -474,6 +475,18 @@ def get_payload(self, query_obj: Optional[QueryObjectDict] = None) -> VizPayload
         filters = self.form_data.get("filters", [])
         filter_columns = [flt.get("col") for flt in filters]
         columns = set(self.datasource.column_names)
+        filter_values_columns = []
+
+        # if using virtual datasource, check filter_values
+        sql = self.datasource.sql
+        if sql:
+            statements_without_comments = sqlparse.format(sql, strip_comments=True)
+            parsed_query = sql_parse.ParsedQuery(statements_without_comments)
+            for stmt in parsed_query.get_statements():
+                filter_values_columns += (
+                    re.findall(r"filter_values\(\"(\w+)\"\,", stmt)
```

Comment:
Maybe make the RegExp a constant? It looks quite random without enough context. Newcomers will have a real hard time understanding this is somehow related to Jinja template.

Reply:
Can we get a comment here explaining what this is doing, too?

Reply:
fixed.